### ISSUE TITLE
Investigate whether we can get gcc headers from toolchain

### ISSUE THREAD

**Issue by alexkalmuk** [embox/embox#2208]
For example, we build GCC for OpenCV to get STL headers, while there is one another way: get the STL library from the toolchain, and get headers from the toolchain. I am not sure about the details (how to get a path to includes) and whether it's possible, but we can investigate this.

p.s.
Another option is to get STL from other places, not from gcc.

**Comment by alexkalmuk** [comment]
We solved this issue in #2208. In short, we have exceptions and STL in this new environment which makes our basic c++ examples (tests, stl example) pass successfully, but there is no std::thread, std::mutex support in STL which makes this solution not possible to apply for Open CV, for instance. 

There is no std::thread support because arm-none-eabi-gcc has `Thread model: single`, which disables threads. So in this solution, we highly rely on the options toolchain was built with.

Another thing is newlib. arm-none-eabi-gcc built with `--with-newlib` which makes libstdc++.a (GCC STL library) have several undefined references to newlib symbols. That's why we added the newlib_compat module.

Finally, libstdc++.a built with wchar support, so we added a bunch on stubs.

As the result, this approach provides C++ support with exceptions and STL, without std::thread support.
